server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+import os
+import sys
+import sh
+import glob
+import signal
+import configtools
+import subprocess
+from argparse import ArgumentParser
+from configparser import ConfigParser, Error as ConfigParserError, NoSectionError, NoOptionError
+
+# Correct the first path to be the "lib" directory to make the
+# pbench-backup-to-s3 script simple, allowing the imports from the
+# backups3 library to work cleanly.
+first_path = sys.path[0]
+assert first_path.endswith('/bin')
+lib_path = os.path.join(os.path.dirname(first_path), 'lib')
+sys.path[0] = lib_path
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+from s3backup import S3Config
+
+_NAME_    = "pbench-backup-tarballs"
+
+# Global logger for the module, setup in main()
+_logger = None
+
+def rsync_archive_backup(config, report):
+    # Function to do rsync between archive and backup
+    exclude = ["--exclude={}".format(i) for i in config.EXCLUDE_DIRS.split(" ")]
+    # N.B The trailing slash is important
+    archive = "{}/".format(config.ARCHIVE)
+    stat_out = sh.rsync("-va", "--stats", exclude, archive, config.BACKUP)
+    if stat_out.exit_code == 0:
+        _logger.info(str(stat_out))
+        report.write(str(stat_out)[-384:])
+    elif stat_out.exit_code != 0:
+        _logger.warning("Failed to do rsync between {} and {}: {}\n".format(config.ARCHIVE, config.BACKUP, e))
+        report.write("Failed to do rsync between {} and {}: {}\n".format(
+            config.ARCHIVE, config.BACKUP, e))
+
+def backup_data_to_s3(s3_obj, config, report):
+    # Function to copy data to s3
+    try:
+        tarlist = glob.glob('{}/*/*.tar.xz'.format(config.ARCHIVE))
```

Comment:
I think this is way too much: it's all the tarballs in the archive - every time. We should be just going over the appropriate linksrc (TO-BACKUP) directory. Similarly in the rsync function. We also need a link destination (BACKED-UP or something) to dump the links into: that's a final state for this processing.